src/coreclr/vm/clsload.cpp

```diff
@@ -2783,6 +2783,12 @@ TypeHandle ClassLoader::PublishType(const TypeKey *pTypeKey, TypeHandle typeHnd)
     }
     CONTRACTL_END;
 
+#ifdef _DEBUG
+    if (!typeHnd.IsTypeDesc())
+    {
+        typeHnd.AsMethodTable()->GetAuxiliaryDataForWrite()->SetIsPublished();
```

Comment:
Why is this only done in a `_DEBUG` build? Normally I would expect an assert or some other clearly debug-esque API being called. Instead `SetIsPublished()` sounds vaguely like a real API that should convey logic. Perhaps a comment?

Reply:
Comment added.

Reply:
// The IsPublished flag is used by various asserts to assure that allocations of
        // MethodTable associated memory which do not use the AllocMemTracker of the MethodTableBuilder
        // aren't permitted until the MethodTable is in a state where the MethodTable object
        // cannot be freed (except by freeing an entire LoaderAllocator)